File: Solutions/1221. Split a String in Balanced Strings/1221. Split a String in Balanced Strings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int res=0,lcnt=0,rcnt=0;
        for(const auto& c:s){
            if(c=='R')
                ++rcnt;
            else if(c=='L')
                ++lcnt;
            if(rcnt!=0 and rcnt==lcnt)
                ++res;
        }
        return res;
    }
};

class Solution {
public:
    int balancedStringSplit(string s) {
        int res=0,cnt=0;
        for(const auto& c:s){
            cnt+=c=='L'?1:-1;
            if(cnt==0)++res;
        }
        return res;
    }
};

File: Solutions/1221. Split a String in Balanced Strings/1221. Split a String in Balanced Strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        res,lcnt,rcnt=0,0,0
        for c in s:
            if c=='R':
                rcnt+=1
            else:
                lcnt+=1
            if rcnt!=0 and rcnt==lcnt:
                res+=1
        return res

class Solution:
    def balancedStringSplit(self, s: str) -> int:
        res,cnt=0,0
        for c in s:
            if c=='L':
                cnt+=1
            else:
                cnt-=1
            if cnt==0:
                res+=1
        return res